packages/dashboard-frontend/src/containers/FactoryLoader/index.tsx

```diff
@@ -316,15 +321,27 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
       workspace = this.props.allWorkspaces.find(workspace => {
         if (isWorkspaceV1(workspace.ref)) {
           // looking for the stack name attribute in Che workspace
-          return workspace.ref.attributes && workspace.ref.attributes.stackName === attrs.stackName;
+          return workspace.ref?.attributes?.stackName === attrs.factoryUrl;
         }
         else {
+          const annotations = workspace.ref.metadata.annotations;
+          const source = annotations ? annotations[DEVWORKSPACE_DEVFILE_SOURCE] : undefined;
+          if (source) {
+            const sourseObj = safeLoad(source) as FactorySource;
+            return sourseObj?.factory?.params === `url=${attrs.factoryUrl}`;
+          }
           // ignore createPolicy for dev workspaces
           return false;
         }
       });
     }
     if (!workspace) {
+      // for backward compatibility with workspaces V1
+      if (isDevfileV1(devfile)) {
+        const { factoryUrl } = attrs;
```

Comment:
```suggestion
        const { factoryUrl } = attrs.factoryUrl;
```
should not it be like ^ ? Or I just miss something in typescript...

Reply:
@sleshchenko it's object destructuring ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring)). It assigns object property value to a variable.

Reply:
OK thanks. Sorry for such a stupid comment

Reply:
don't say that, and keep asking :)